images/extension-builders/tinygo/commands.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Tetrate
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+extension_build()  {
+	tinygo build -o "$1" -scheduler=none -target wasi main.go
+	# This is necessary since the created go caches are with read-only permission,
+	# and without this, the host user cannot delete the build directory with "rm -rf".
+	chmod -R a+rw build/
```

Comment:
Giving RW permissions to everyone looks scary on my Mac:

<img width="477" alt="Screenshot 2021-03-12 at 10 17 25" src="https://user-images.githubusercontent.com/19395458/110919084-25b6ec00-831c-11eb-8b96-48889395449f.png">

Let's avoid this kind of questions (from security-focused people) by narrowing down the scope of permissions:

```shell
chmod -R u+w "${GOMODCACHE}"
```

Reply:
done https://github.com/tetratelabs/getenvoy/pull/104/commits/db99b4af2ff676b7ea0dd6c87a2a1d03b7f3e30f :+1: